### DISCUSSION TITLE
extend borderWidth

### DISCUSSION THREAD

**Discussion by Hakira-Shymuy** [tailwindlabs/tailwindcss#5026] [Help] [answered]
hi guys, i have the following situation

`extend:{
borderWidth: ['hover'],
}`

and on terminal it gives me a Tailwind CSS: VariantsValue is not iterable

however its working as i wanted

any thoughs?

**Comment by olets** [comment]
Looks like you're trying to add support for the hover variant of border-width. That's `variants` not `extend` https://play.tailwindcss.com/ezEU5WUTS8?file=config

However, that mistake on its own shouldn't give you that error. Instead you should get (unhelpful) styles like 

```css
.border-0 {
  border-width: hover;
}
```

For more see

https://tailwindcss.com/docs/border-width#customizing

**Reply by Hakira-Shymuy** [reply]
hi, thanks for the answer
what im trying to do is the same as example bellow from simonswiss
my config have the same way he have
and, it is actually working, however have no idea why add the error

right now i have no errors on my console

is it possible that vscode can have some problems with some sync?
because some times after i save a doc, html, js or config it gives me something on the problems tab, and another times is completely okay

**Comment by simonswiss** [comment, marked as answer]
Hey!

Can you share your full Tailwind config? Are you using `extend` inside the `variants` key? If so, that should work properly.

Are you using the JIT mode? If so, you don't need to extend variants, they are _all_ enabled by default.

Here's a Tailwind Play demo (for the non JIT version):

https://play.tailwindcss.com/zk9yCWYkQd?file=config

https://user-images.githubusercontent.com/485747/126595040-2e563b52-1a5f-46a2-97e9-454df7c5a52b.mov

**Reply by Hakira-Shymuy** [reply]
```
module.exports = {
  purge: [],
  darkMode: false, // or 'media' or 'class'
  theme: {
    extend: {
      backgroundImage: theme => ({
        'hero-section': "url('img/dummy.jpg')",
      }),
      fontFamily: {
        title: ['Barlow'],
        body: ['Lato']
      },
    },
  },
  variants: {
    extend: {
      textColor: ['group-hover'],
      borderWidth: ['hover'],
    },
  },
  plugins: [],
}
```

**Reply by Hakira-Shymuy** [reply]
thx for your answer, at moment i have no problems on my vscode tab, but some times it shows.
from what i see, i have the config on the right way, and actually its working as i wanted, no idea why it was reporting a problem
